### DISCUSSION TITLE
More features around insert

### DISCUSSION THREAD

**Discussion by Panaetius** [ahopkins/mayim#42] [Ideas]
"Insert" isn't mentioned once in the docs right now, and while support for a simple one row insert is there, one has to fall back to https://ahopkins.github.io/mayim/guide/executors.html#low-level-run-sql for anything beyond that.

The two things I'd love to see (other than some examples on inserts for people with less experience) is:

- Inserting multiple values by passing a list, e.g. something like
```
class Entry(NamedTuple):
    id: int
    name: str

class Executor(...):
    @query("""
        INSERT INTO entries (id, name) VALUES $values[($id, $name)];
    """)
    async def mayim_insert(values: list[Entry])-> None:
        ...
```
producing something like
```
INSERT INTO entries (id, name) VALUES (1, "a"),(2, "b"),(3,"c");
```
- The above to also somehow work with nested objects/pydantic, i.e. in combination with https://github.com/ahopkins/mayim/issues/41 . though it's not clear to me how a clean syntax for this would look like

**Comment by ahopkins** [comment]
This is an excellent idea. Will need to think what the ideal way to implement something like this is, but that certainly makes sense. I suppose the question is how to differentiate what is to be multiple sets of values from what is supposed to be an array SQL type.

**Reply by Panaetius** [reply]
Yes, the syntax is what I'm not clear on, there's lots of room to get it "wrong".
The above suggestion was the first idea I had. But of course it could also be something more like in Jinja and other templating languages with a `for` construct. Though then just using one of those template libraries with a syntax already familiar to user might make sense.

**Reply by ahopkins** [reply]
I'm not opposed to your syntax. I really don't want to go the route of making a whole templating language introducing loops, etc. Keeping it close to what python devs already know with type annotations make sense.
